include/seqan/sequence/string_set_owner.h

```diff
@@ -141,6 +141,21 @@ class StringSet<TString, Owner<Default> >
 // ============================================================================
 
 // --------------------------------------------------------------------------
+// Function append()
+// --------------------------------------------------------------------------
+
+// better solution if both stringsets are Owner<Default>
+template <typename TString, typename TString2, typename TExpand >
+inline void append(StringSet<TString, Owner<Default> > & me,
+                   StringSet<TString2, Owner<Default> > const & obj,
+                   Tag<TExpand>)
+{
+    // we rather invalidate limits here to allow to do modify appended strings:
+    me.limitsValid = false;
+    append(me.strings, obj.strings, Tag<TExpand>());
+}
+
+// --------------------------------------------------------------------------
```

Comment:
Why do we need this overload?
This is essentially the same as the generic append version for string sets.
The difference is that limits is not valid, but we also have dependent string sets, which do need to set their limits accordingly. I think it would be better to add a helper function setLimitsState which does nothing for STL containers and otherwise sets the limits to ```false```.